library/alloc/src/collections/binary_heap.rs

```diff
@@ -495,7 +495,10 @@ impl<T: Ord> BinaryHeap<T> {
         let mut end = self.len();
         while end > 1 {
             end -= 1;
-            self.data.swap(0, end);
+            unsafe {
```

Comment:
Let's add a comment here of the form `// SAFETY: ..` to indicate why this `unsafe` block is actually safe.